pkg/pillar/cmd/zedrouter/appnumonunet.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Network Instance/underlay network IP Address Management,
+// App Number management Module
+
+package zedrouter
+
+import (
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/uuidtonum"
+	"github.com/satori/go.uuid"
+)
+
+// mapped on base UUID
+var appNumBase map[string]*types.AppNumBase
+
+// returns base bitMap for a given UUID
+func appNumBaseGet(baseID uuid.UUID) *types.AppNumBase {
+	if appNumBase != nil {
+		uuidStr := baseID.String()
+		if baseMap, ok := appNumBase[uuidStr]; ok {
+			return baseMap
+		}
+	}
+	return nil
+}
+
+// Create application number Base for a given UUID
+func appNumBaseCreate(ctx *zedrouterContext, baseID uuid.UUID, size int) bool {
+	if appNumBase == nil {
+		appNumBase = make(map[string]*types.AppNumBase)
+	} else {
+		// already created
+		if appNumBaseGet(baseID) != nil {
+			return false
+		}
+	}
+	log.Functionf("appNumBaseCreate (%s)\n", baseID.String())
+	baseMap := new(types.AppNumBase)
+	appNumBase[baseID.String()] = baseMap
+	appNumsOnNetworkInstanceRestore(ctx, baseMap, baseID, size)
+	return true
+}
+
+func appNumBaseDelete(ctx *zedrouterContext, baseID uuid.UUID) {
+	baseMap := appNumBaseGet(baseID)
+	if baseMap != nil {
+		appNumsOnNetworkInstanceDelete(ctx, baseMap, baseID)
+		delete(appNumBase, baseID.String())
+	}
+}
+
+// AppNumber garbage collector: for non-existent network instance/
+// app instance, free the persistent appNum map
+func appNumMapGC(ctx *zedrouterContext) {
+	log.Functionf("persistentAppNumMapGC")
+	freedCount := 0
+	pub := ctx.pubUuidsToNum
+	items := pub.GetAll()
+	for _, item := range items {
+		appNumMap := item.(types.UuidsToNum)
+		appID := appNumMap.AppID
+		baseID := appNumMap.BaseID
+		appNetStatus := lookupAppNetworkStatus(ctx, appID.String())
+		netInstStatus := lookupNetworkInstanceStatus(ctx, baseID.String())
+		if netInstStatus == nil || appNetStatus == nil {
+			log.Functionf("persistentAppNumMapGC: freeing %s", appNumMap.Key())
+			appNumOnUNetDelete(ctx, baseID, appID, false)
+			freedCount++
+		}
+	}
+	log.Functionf("persistentAppNumMapGC freed %d", freedCount)
+}
+
+// Allocate AppNum for a given pair of UUIDs
+func appNumOnUNetAlloc(ctx *zedrouterContext, baseID uuid.UUID,
+	appID uuid.UUID, baseSize int, reset bool) (int, error) {
+	pub := ctx.pubUuidsToNum
+	numType := "appNumOnUnet"
+	baseMap := appNumBaseGet(baseID)
+	if baseMap == nil || baseSize == 0 {
+		log.Functionf("appNumAlloc(%s): not Addressable\n",
+			baseID.String())
+		return types.AppNumInvalid, nil
+	}
```

Comment:
What does this map add? UuidsToNumGet also looks up in a map and having two means more complexity and more bugs.